genie-core/src/main/java/com/netflix/genie/core/services/impl/CacheGenieFileTransferService.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.netflix.genie.core.services.impl;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.common.exceptions.GenieServerException;
+import com.netflix.genie.core.services.FileTransfer;
+import com.netflix.genie.core.services.FileTransferFactory;
+import com.netflix.spectator.api.Registry;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.validator.constraints.NotBlank;
+
+import javax.validation.constraints.NotNull;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+import java.util.function.ToDoubleFunction;
+
+/**
+ * Caches the downloaded file from the remote location.
+ * Created by amajumdar on 7/22/16.
+ */
+@Slf4j
+public class CacheGenieFileTransferService extends GenieFileTransferService {
+    //File cache location
+    private final String baseCacheLocation;
+    //File transfer service to get/put files on a local system
+    private final FileTransfer localFileTransfer;
+    //File cache
+    private final LoadingCache<String, File> fileCache = CacheBuilder.newBuilder()
```

Comment:
Nice I didn't know such a thing existed. We can possibly use this for on node caching of security tokens with a TTL if we want to get around the current bottleneck quickly.